Source/ASCollectionView.mm

```diff
@@ -1930,6 +1972,11 @@ - (void)layer:(CALayer *)layer didChangeBoundsWithOldValue:(CGRect)oldBounds new
   if (self.collectionViewLayout == nil) {
     return;
   }
+  if (_collectionViewLayoutFlags.providesLayoutCalculator) {
+    // TODO Maybe kick off a background calculation here and hope that the layout is ready by the time -prepareLayout
```

Comment:
Agreed, definitely not gonna work on this in this PR, if ever. It was an idea I thought might worth considering. Gonna remove the TODO for now.